1_IGT_PP/Code/R/0_Preprocess/SelfReportPreprocess.R

```diff
@@ -0,0 +1,169 @@
+                                   "panas_na", "masqGDA", "masqAA", "masqGDD",
+                                   "masqAD", "shaps_tot", "prdep_tot")) %>% 
+  mutate(session = 1) %>%                     # Create session variable
+  filter(ID >= 2049,                          # Subset PP participants
+         ID != 2059,                          # Remove participant that played old IFT
+         ID != 2083)                          # Remove participant that had no task data
+
+session2 = read_sav(here("1_IGT_PP", "Data", "0_Raw", "MergedQuest_3.21.16-Session2.sav"),
+                    col_select = c("ID", "bastot", "basdrive", "basfunsk",
+                                   "basrewres", "bis", "panas_pa",
+                                   "panas_na", "shaps_tot")) %>% 
+  mutate(session = 2) %>%                     # Create session variable
+  filter(ID >= 2049,                          # Subset PP participants
+         ID != 2059,                          # Remove participant that played old IFT
+         ID != 2083,                          # Remove participant that had no task data
+         !ID %in% c(2057, 2063, 2064,         # Remove participants that didn't do session 2
+                    2067, 2086, 2090,
+                    2093, 2094, 2096, 2098))
+
+both_sessions = bind_rows(session1, session2) %>% 
+  select(ID, session,
+         # Session 1 only
+         prdep_tot, masqGDA, masqAA, masqGDD, masqAD,
+         # Both sessions
+         bastot, basdrive, basfunsk, basrewres,
+         bis, panas_pa, panas_na, shaps_tot,
+         # Remove demographics
+         -c(sex, race, Ethnicity)) %>% 
+  pivot_longer(cols = -c("ID", "session"), values_to = "score", names_to = "scale")
+
+  
+  
+# -----------------------------------------------
+# PREP DATA
+IDs = unique(both_sessions$ID)
+N = length(IDs)
+
+# Make dataframe describing each scale
+scales = data.frame(scale = c(# Session 1 only
+                              "prdep_tot", "masqGDA", "masqAA", "masqGDD", "masqAD",
+                              # Both Sessions
+                              "bastot", "basdrive", "basfunsk", "basrewres",
+                              "bis", "panas_pa", "panas_na", "shaps_tot"),
+                    n_items = c(28, 11, 17, 12, NA_real_, 13, 4, 4, 5, 7, 10, 10, 14),
+                    min_item_score = c(0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1),
+                    max_item_score = c(4, 5, 5, 5, NA_real_, 4, 4, 4, 4, 4, 5, 5, 4),
+                    n_sessions = c(1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2)) %>% 
+  mutate(min = n_items * min_item_score,
+         max = n_items * max_item_score,
+         scaled_max = max - min)
+
+# Make scaled scores and check out which scores are below the minimums - for those scores, we'll truncate them to 0
+full_data = both_sessions %>% 
+  left_join(scales) %>% 
+  mutate(scaled_score = case_when(score-min < 0 ~ 0, is.numeric(score) ~ score - min))
```

Comment:
I think scaling makes sense! For the binomial model, we would just want to make sure that each response increment is accounted for. e.g., answering the lowest on all items should give a 0, and answering the highest on all items should give a number that indicates each possible increment above 0. Basically, the first category for each item should be scored as 0, and every ordinal category above the first category should be scored with an additional +1 (let me know if I am explaining that sensibly ha)

Reply:
That makes sense!... I'm pretty sure haha. Essentially, the participants' scores should be bounded between 0% (answering the lowest on all items) and 100% (answering the highest on all items) where 100% is entered in the model as the actual maximum score for that scale. The code above should make those adjustments. I still have to double-check some of those. Also, some participants had missing items, which put their score below 0% in some cases. For now, I truncated those to 0 which assumes they scored 0 on the missing item(s) which seemed reasonable to me (they scored '0' on all the other items). Let me know if you have other thoughts. I'm also going to bring this up with Tom today. Thanks!